core/common/src/main/java/alluxio/util/io/FileUtils.java

```diff
@@ -259,18 +256,22 @@ public FileVisitResult postVisitDirectory(Path dir, IOException e) throws IOExce
    * @param path storage directory path to create
    */
   public static void createStorageDirPath(String path) throws IOException {
-    File dir = new File(path);
-    if (dir.exists()) {
+    if (Files.exists(Paths.get(path))) {
       return;
     }
-    if (!dir.mkdirs()) {
-      if (dir.exists()) {
-        // This dir has been created concurrently.
-        return;
-      }
+    Path storagePath;
+    try {
+      storagePath = Files.createDirectories(Paths.get(path));
+    } catch (FileAlreadyExistsException e1) {
```

Comment:
with Java 7+, you can combine the following catch blocks into one like this. Also it is good to put the original exception `e` here as the cause for the new exception thrown.
```java
} catch (AlluxioException | IOException e) {
  throw new IOException("Failed to create folder " + path, e);
}
```